### PR TITLE
[MM-55] Update plugin with respect to phase 1 upgrades

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Updated plugin with respect to phase 1 upgrades

- Updated Makefile
- Updated build folder
- Updated version in `nvmrc` file and update `package-lock.json` file.
- Updated developer docs link in the readme file 
- Updated package in go modules to use https://pkg.go.dev/github.com/mattermost/mattermost/server/public.
- Added typescript setup

#### Ticket Link

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,11 +221,7 @@ You can also start a meeting in any chat window by typing `/zoom start`.
 
 ## Development
 
-This plugin contains both a server and web app portion. 
-
-Use `make dist` to build distributions of the plugin that you can upload to a Mattermost server for testing.
-
-Use `make check-style` to check the style for the whole plugin.
+This plugin contains both a server and web app portion. Read our documentation about the [Developer Workflow](https://developers.mattermost.com/integrate/plugins/developer-workflow/) and [Developer Setup](https://developers.mattermost.com/integrate/plugins/developer-setup/) for more information about developing and extending plugins.
```
Comment (mickmister) [inline README.md:225]: Awesome :+1:

File: build/manifest/main.go [hunk 2]
```diff
@@ -82,7 +57,7 @@ func findManifest() (*model.Manifest, error) {
 	// we don't want to accidentally clobber anything we won't preserve.
 	var manifest model.Manifest
 	decoder := json.NewDecoder(manifestFile)
-	// decoder.DisallowUnknownFields() // Commented out until circleci image is updated: https://github.com/mattermost/mattermost-plugin-zoom/pull/269#discussion_r927075701
+	decoder.DisallowUnknownFields()
```
Comment (mickmister) [inline build/manifest/main.go:60]: :+1:

File: server/plugin_test.go [hunk 3]
```diff
@@ -142,6 +142,15 @@ func TestPlugin(t *testing.T) {
...
 			api.On("KVSetWithExpiry", "zoomuserstate_theuserid", mock.AnythingOfType("[]uint8"), mock.AnythingOfType("int64")).Return(nil)
 
+			api.On("KVSetWithOptions", "mutex_mmi_bot_ensure", mock.AnythingOfType("[]uint8"), model.PluginKVSetOptions{Atomic: true, OldValue: []uint8(nil), ExpireInSeconds: 15}).Return(true, nil)
+			api.On("KVSetWithOptions", "mutex_mmi_bot_ensure", []byte(nil), model.PluginKVSetOptions{ExpireInSeconds: 0}).Return(true, nil)
+
+			api.On("EnsureBotUser", &model.Bot{
+				Username:    botUserName,
+				DisplayName: botDisplayName,
+				Description: botDescription,
+			}).Return(botUserID, nil)
```
Comment (mickmister) [inline server/plugin_test.go:152]: Interesting

**Review by mickmister — APPROVED** [review]

File: server/plugin_test.go (hunk 3 above)
Comment (ayusht2810) [reply server/plugin_test.go:152]: @mickmister What risk might you be talking about on this?

**Comment by mickmister** [comment]
Hi @ayusht2810, can you please resolve the conflicts here? Thank you

**Review by mickmister — APPROVED** [review]

**Pull request merged**